packages/d3fc-webgl/src/buffers/vertexConstantAttributeBuilder.js

```diff
@@ -0,0 +1,75 @@
+import baseAttributeBuilder from './baseAttributeBuilder';
+import { rebindAll, exclude } from '@d3fc/d3fc-rebind';
+
+export default () => {
+    const base = baseAttributeBuilder();
+
+    let data = null;
+    let validSize = 0;
+
+    const project = (elementCount, verticesPerElement) => {
+        const components = base.size();
+        const offset = base.offset();
+        const elementComponentCount = verticesPerElement * components;
+        const projectedData = new Float32Array(
+            offset + elementCount * elementComponentCount
+        );
+        if (data.length !== verticesPerElement) {
+            throw new Error(
+                `Expected vertices array of size ${verticesPerElement}, recieved array with length ${data.length}.`
+            );
+        }
+        let target = 0;
+        for (let vertex = 0; vertex < verticesPerElement; vertex++) {
+            if (data[vertex].length !== components) {
+                throw new Error(
+                    `Expected components array of size ${components}, recieved array with length ${data[vertex].length}.`
+                );
+            }
+            for (let component = 0; component < components; component++) {
+                projectedData[target] = data[vertex][component];
```

Comment:
Should this be `projectedData[offset + target]`?

Reply:
I reckon so, done.